### ISSUE TITLE
macOS Vault Master Password Entry Holds Secure Keyboard Input Open

### ISSUE THREAD

**Issue by mwegner** [bitwarden/desktop#298]
With the macOS Desktop Vault, version 1.15.2, I'm seeing some new behavior with secure keyboard input or keyboard focus.  I'm on Mojave 10.14.6.

My issue is that Bitwarden keeps keyboard focus active in the Master Password field, even when it's in the background.  When secure keyboard input is active, system-wide keyboard macro software is unable to function (in my case, Keyboard Maestro).

This is a recent break for me, but I'm not sure if secure keyboard input is new, or focus behavior has changed.  As near as I can tell, here's what's happening:

- When Bitwarden loses application focus, it ends the secure input/keyboard focus on the password field

- When Bitwarden regains application focus, it puts keyboard focus back on the field (if it had it when you left; if you click off the master password field to stop keyboard focus, it won't return it when application focus returns)

This is all fine!

But:  It takes a few seconds for Bitwarden to go from application start to fully initialized, where it renders the vault locked UI.  _If it loses application focus during this window_, it still begins with keyboard focus in the master password entry, and thus secure input, even though it's no longer the front application.

The way my applications start up on boot, by the time Bitwarden has fully initialized, some other restarting application has taken focus.  I have to tab in and out of Bitwarden, or manually click off the master password entry text field, before Keyboard Maestro begins working.

I think that adding a check to see if Bitwarden is still the front-most application before putting keyboard focus into the password field will fix this.

**Comment by kspearrin** [comment]
The recent update upgraded our version of Electron, so maybe something internally changed there. We don't do any checks to see if Bitwarden is the foregrounded applications, so I would need to look into Electron APIs to see if that is possible to detect.

**Comment by mtgto** [comment]
Good news. You can now use `Secure Keyboard Input` by Electron v10 or later.
https://github.com/electron/electron/blob/v10.0.0/docs/api/app.md#appissecurekeyboardentryenabled-macos

**Comment by joejoinerr** [comment]
+1 for implementing this! It's playing havoc with my KM macros and text expansion. I thought it was a macOS issue but found that whenever I quit Bitwarden I never have the issue.

**Comment by danmandle** [comment]
This is still an ongoing issue. There is another issue here:  #738

**Comment by eliykat** [comment]
There are several issues reporting the same problem with different apps. I'm going to close all the other threads and use this as the main issue. However I'll link them here for reference:
* #700 and #738: Alfred text expansion app
* #968: Adobe Indesign hotkeys
* #921: Keyboard Maestro

**Comment by Dasymi** [comment]
I've got the same issue with Bitwarden preventing text expansion by Alfred. Is there any update on this?

**Comment by eliykat** [comment]
Hi @Dasymi, this bug should be fixed by upgrading to the latest version of Bitwarden desktop (v 1.27.1). If you're still experiencing this bug after upgrading, please post here with more info about how to reproduce.

**Comment by xeroxoid** [comment]
Hi @eliykat. I see [v1.28.2](https://github.com/bitwarden/desktop/releases/tag/v1.28.2) has been released a couple of weeks ago. Every when are new versions released to the Mac App Store?

I am still on v1.27.1 (531) via the App Store just because I am using the unlock with biometrics for Chrome Extension functionality which as I understand is only available with Bitwarden through the App Store.

**Comment by eliykat** [comment]
Hi @xeroxoid, we submit our releases to the app store at the same time they are released on our website, however the approval process with Apple can take a few weeks. If it's been 2 weeks already then hopefully it's not too far away.

**Comment by eliykat** [comment]
@xeroxoid, v1.28.2 is now live on the Mac App Store.

**Comment by xeroxoid** [comment]
TY for the update!

**Comment by eliykat** [comment]
Hi @escobar5, can you please:
1. describe what problem you're having
2. provide step-by-step instructions on how to reproduce the problem
